File: 17solve/17-158.py
# Автор В.Н. Шубинкин

with open('17-1.txt') as f:
    a = [int(x) for x in f.readlines()]

max_len = 0
count = 0
k = 1
for i in range(1, len(a)):
    if a[i] < a[i - 1]:
        k += 1
        if k > max_len:
            max_len = k
            count = 0
        if k == max_len:
            count += 1
    else:
        k = 1
print(max_len, count)

File: 17solve/17-158.pas
// Автор В.Н. Шубинкин
##
var a := ReadLines('17-1.txt').Select(t -> t.ToInteger).ToArray;
var maxLen := 0;
var Len := 1;
var count := 0;
for var i := 1 to a.High do
begin
  if a[i] < a[i - 1] then
  begin
    Len += 1;
    if Len > maxLen then
    begin
      maxLen := Len;
      count := 0
    end;
    if Len = maxLen then count += 1
  end
  else Len := 1;
end;
print(maxLen, count)